src/duckling/canvas/tools/entity-move-tool.ts

```diff
@@ -110,27 +117,30 @@ export class EntityMoveTool extends BaseTool {
         this._selectionService.deselect(mergeKey);
         this._entitySystemService.deleteEntity(entityKey, mergeKey);
     }
-
-    private _buildSelectionBox(canvasZoom : number) : DisplayObject {
-        let graphics : Graphics = null;
-        let selectedEntityKey = this._selectionService.selection.value.selectedEntity;
-        let selectedEntity = this._entitySystemService.getEntity(selectedEntityKey);
-        if (selectedEntity) {
-            graphics = this._buildSelectionBoxAroundEntity(selectedEntity, canvasZoom);
-        }
-        return graphics;
+    
+    private _cancel() {
+        this._selection = null;
     }
 
-    private _buildSelectionBoxAroundEntity(entity : Entity, canvasZoom : number) : Graphics {
+    private _drawSelectionBox(box : Box2, canvasZoom : number) : Graphics {
         let graphics : Graphics = null;
-        let box = this._entityBoxService.getEntityBox(entity);
         if (box) {
             graphics = new Graphics();
             graphics.lineStyle(1 / canvasZoom, 0x3355cc, 1);
             drawRectangle(box.position, box.dimension, graphics);
         }
         return graphics;
     }
+    
+    private _getSelectedEntityBox() {
+        let selectedEntityKey = this._selectionService.selection.value.selectedEntity;
```

Comment:
These two lines could be simplified  to this._selectionService.selection.value.entity;